solver/src/main/java/org/chocosolver/solver/variables/view/integer/IntAffineView.java

```diff
@@ -0,0 +1,588 @@
+
+        if (p) {
+            v = var.nextValue(v);
+        } else {
+            v = var.previousValue(v);
+        }
+        if (v == Integer.MIN_VALUE || v == Integer.MAX_VALUE) return Integer.MAX_VALUE;
+        return (p ? v : -v) * a + b;
+    }
+
+    @Override
+    public int nextValueOut(int v) {
+        v++;
+        if (v < getLB() || v > getUB()) return v;
+        // y = p * a * x + b where p in {-1,1}, a and b in Z.
+        // => x = (y - b) / (p * a)
+        double w = v - b;
+        w /= (p ? a : -a);
+
+        int k = (int) w;
+        if (w == k && var.contains(k)) {
+            if (a > 1) {
+                v++;
+            } else {
+                if (p) {
+                    k = var.nextValueOut(k);
+                } else {
+                    k = var.previousValueOut(k);
+                }
+                v = ((p ? k : -k) * a + b);
+            }
+        }
+        return v;
+    }
+
+    @Override
+    public int previousValue(int v) {
+        if (v > getUB()) return getUB();
+        if (v < getLB()) return Integer.MIN_VALUE;
+        // y = p * a * x + b where p in {-1,1}, a and b in Z.
+        // => x = (y - b) / (p * a)
+        v -= b;
+        v = (v > 0 && a > 1) ? v + 1 : v;
+        v /= (p ? a : -a);
+
+        if (p) {
+            v = var.previousValue(v);
+        } else {
+            v = var.nextValue(v);
+        }
+        if (v == Integer.MIN_VALUE || v == Integer.MAX_VALUE) return Integer.MIN_VALUE;
+        return (p ? v : -v) * a + b;
+    }
+
+    @Override
+    public int previousValueOut(int v) {
+        v--;
+        if (v < getLB() || v > getUB()) return v;
+        // y = p * a * x + b where p in {-1,1}, a and b in Z.
+        // => x = (y - b) / (p * a)
+        double w = v - b;
+        w /= (p ? a : -a);
+
+        int k = (int) w;
+        if (w == k && var.contains(k)) {
+            if (a > 1) {
+                v--;
+            } else {
+                if (p) {
+                    k = var.previousValueOut(k);
+                } else {
+                    k = var.nextValueOut(k);
+                }
+                v = ((p ? k : -k) * a + b);
+            }
+        }
+        return v;
+    }
+
+
+    @Override
+    protected EvtScheduler<IntEventType> createScheduler() {
+        return new IntEvtScheduler();
+    }
+
+    @Override
+    public String toString() {
+        return this.getName() + "[" + getLB() + "," + getUB() + "]";
+    }
+
+    @Override
+    public IEventType transformEvent(IEventType evt) {
+        if (evt == INCLOW) {
+            if (!p) return DECUPP;
+        } else if (evt == DECUPP) {
+            if (!p) return INCLOW;
+        }
+        return evt;
+    }
+
+    @Override
+    public boolean hasEnumeratedDomain() {
+        return var.hasEnumeratedDomain() || a > 1;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof IntAffineView)) return false;
```

Comment:
I make this remark here, but it stands for the entire class : I am not a fan of inline if, for, while, and so on (but some standard formats allow them for short lines such as these ones). I know that we did not choose formatting rules for the code in Choco, so it is just a remark, you might leave it this way.